noir-projects/noir-contracts/contracts/nft_contract/src/main.nr

```diff
@@ -0,0 +1,191 @@
+// Minimal NFT implementation with `AuthWit` support that allows minting in public-only and transfers in both public
+// and private.
+contract NFT {
+    #[aztec(event)]
+    struct Transfer {
+        from: AztecAddress,
+        to: AztecAddress,
+        token_id: Field,
+    }
+
+    #[aztec(storage)]
+    struct Storage {
+        // The symbol of the NFT
+        symbol: SharedImmutable<FieldCompressedString>,
+        // The name of the NFT
+        name: SharedImmutable<FieldCompressedString>,
```

Comment:
Packing multiple constants in 1 struct generally makes sense to me I am just not sure if it makes sense in this context because I would say these values will never be read in an actual tx (only read via getters for offchain purposes).